File: js_implement/quick_review_2025/125.验证回文串.js
/*
 * @lc app=leetcode.cn id=125 lang=javascript
 * @lcpr version=30005
 *
 * [125] 验证回文串
 */


// @lcpr-template-start

// @lcpr-template-end
// @lc code=start
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
  const regxp = /\d|[a-zA-Z]/
  let res = s.toLowerCase().split('').filter(c =>
    regxp.test(c)
  )
  return res.join('') === res.reverse().join('')
};
// @lc code=end



/*
// @lcpr case=start
// "A man, a plan, a canal: Panama"\n
// @lcpr case=end

// @lcpr case=start
// "race a car"\n
// @lcpr case=end

// @lcpr case=start
// " "\n
// @lcpr case=end

 */

File: python_implement/125.验证回文串.py
#
# @lc app=leetcode.cn id=125 lang=python3
# @lcpr version=30404
#
# [125] 验证回文串
#

# @lc code=start
class Solution:
    def isPalindrome(self, s: str) -> bool:
        sb = []
        for c in s:
            if c.isalnum():
                sb.append(c.lower())
        s = ''.join(sb)
        left, right = 0, len(s)-1
        while left < right:
            if s[left]!=s[right]:
                return False
            left+=1
            right-=1
        return True
# @lc code=end



#
# @lcpr case=start
# "A man, a plan, a canal: Panama"\n
# @lcpr case=end

# @lcpr case=start
# "race a car"\n
# @lcpr case=end

# @lcpr case=start
# " "\n
# @lcpr case=end

#

